examples/collinear_magnetism.jl

```diff
@@ -0,0 +1,92 @@
+# # Collinear spin and magnetic systems
+#
+# In this example we consider iron in the BCC phase.
+# To show that this material is ferromagnetic we will model it once
+# allowing collinear spin polarization and once without
+# and compare the resulting SCF energies. In particular
+# the ground state can only be found if collinear spins are allowed.
+#
+# First we setup BCC iron without spin polarization
+# using a single iron atom inside the unit cell.
+using DFTK
+
+a = 5.42352  # Bohr
+lattice = a / 2 * [[-1  1  1];
+                   [ 1 -1  1];
+                   [ 1  1 -1]]
+Fe = ElementPsp(:Fe, psp=load_psp("hgh/lda/Fe-q8.hgh"))
+atoms = [Fe => [zeros(3)]];
+
+# To get the ground-state energy we use an LDA model and rather moderate
+# discretisation parameters.
+
+kgrid = [3, 3, 3]  # k-point grid (Regular Monkhorst-Pack grid)
+Ecut = 15          # kinetic energy cutoff in Hartree
+model_nospin = model_LDA(lattice, atoms, temperature=0.01)
+basis_nospin = PlaneWaveBasis(model_nospin, Ecut; kgrid=kgrid)
+
+scfres_nospin = self_consistent_field(basis_nospin, tol=1e-6, mixing=KerkerMixing())
+
+scfres_nospin.energies
+
+# Since we did not specify any initial magnetic moment on the iron atom,
+# DFTK will automatically assume that a calculation with only spin-paired
+# electrons should be performed. As a result the obtained ground state
+# features no spin-polarization.
+
+# Now we repeat the calculation, but give the iron atom an initial magnetic moment.
+# For specifying the magnetic moment DFTK uses units of the Bohr magneton ``μ_B``,
+# which in atomic units has the value ``\frac{1}{2}`` and is the magnetic moment
+# of a single electron. As a result the value to pass to the model and the guess density
+# function agrees with the desired excess of spin-up over spin-down electrons
+# at the iron centre. In this case we seek the case with as many spin-parallel
```

Comment:
Still very confusing, at least to me... We can't say that DFTK uses the Bohr magneton, since it uses atomic units. Can't we say that DFTK uses the convention that the magnetic moment is the excess of spin-up over spin-down electrons, and leave it to people to figure out what this means in terms of SI units if they want to? In any case we should begin with the "excess" definition since that's both the more convenient one to visualize and what is used computationally.

Reply:
Agree, but the thing with the Bohr magneton is still worth mentioning, but maybe it should be more of a side remark.

Reply:
Agreed, but it should come after the "computational" definition.